packages/magic-entries-webpack-plugin/src/test/utilities.ts

```diff
@@ -0,0 +1,48 @@
+import * as path from 'path';
+
+import * as fs from 'fs-extra';
+
+export class Workspace {
+  constructor(public readonly root: string) {}
+
+  resolvePath(...parts: string[]) {
+    return path.resolve(this.root, ...parts);
+  }
+
+  buildPath(...parts: string[]) {
+    return this.resolvePath('build', ...parts);
+  }
+
+  async write(file: string, contents: string) {
+    const filePath = this.resolvePath(file);
+    await fs.mkdirp(path.dirname(filePath));
+    await fs.writeFile(filePath, contents);
+  }
+
+  async cleanup() {
+    await fs.emptyDir(this.root);
+    await fs.remove(this.root);
+  }
+}
+
+const rootFixtureDirectory = path.resolve(__dirname, '../fixtures');
+
+export async function createWorkspace({name}: {name: string}) {
+  const fixtureDirectory = path.join(rootFixtureDirectory, name);
+  await fs.mkdirp(fixtureDirectory);
```

Comment:
Can `mkdirp` be done in the `Workspace` constructor instead?

Reply:
Looking at it again, we could do that but if we did we'd need to use sync versions of everything which is bad especially since with tests things often want to run in parallel. I suppose we could move `createWorkspace` onto `Workspace.create` which might be a bit nicer.